### PR TITLE
[Feature/Extension] Rename the term 'extension' into 'on_behalf_of'

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Rename the term 'extension' into 'on_behalf_of'.
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Bug fix

### Issues Resolved
* Resolve https://github.com/opensearch-project/security/issues/2760
* Relate https://github.com/opensearch-project/security/pull/2671
* Relate https://github.com/opensearch-project/security/pull/2567

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by RyanL1997** [comment]
Will fix the 10s timeout error by sync with main branch.

File: src/integrationTest/java/org/opensearch/security/SecurityConfigurationTests.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ public void shouldUseSecurityAdminTool() throws Exception {
 	}
 
 	@Test
-	public void shouldReloadExtensionsConfigurationFromFile() throws Exception {
+	public void shouldReloadOnBehalfOfConfigurationFromFile() throws Exception {
```
Comment (DarshitChanpura) [inline src/integrationTest/java/org/opensearch/security/SecurityConfigurationTests.java:218]: seems a little odd reading it at first, but will get used to it :D

**Review by DarshitChanpura — APPROVED** [review]
Codecov failures are out of scope for this PR and hence this one can be merged bypassing the Codecov check once other checks pass.

**Review by cwperks — APPROVED** [review]

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java [hunk 2]
```diff
@@ -847,7 +847,7 @@ public Collection<Object> createComponents(Client localClient, ClusterService cl
 
         securityRestHandler = new SecurityRestFilter(backendRegistry, auditLog, threadPool,
                 principalExtractor, settings, configPath, compatConfig);
-        //TODO: CREATE A INSTANCE OF HTTPExtensionAuthenticationBackend
+        //TODO: CREATE A INSTANCE OF HTTPOnBehalfOfAuthenticationBackend
```
[the remark this answers is not in the document]
Comment (RyanL1997) [reply src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:850]: Sure~

**Pull request merged**